pulsar/internal/connection.go

```diff
@@ -129,9 +134,9 @@ type connection struct {
 	requestIDGenerator uint64
 
 	incomingRequestsCh chan *request
+	incomingCmdCh     chan *incomingCmd
 	writeRequestsCh    chan []byte
 
-	mapMutex    sync.RWMutex
 	pendingReqs map[uint64]*request
```

Comment:
Maybe add a comment on how this should only be accessed in certain places?

Reply:
Agree with @cckellogg 

> The pendingReqs was really meant to be accessed by a single go-routine.

`pendingReqs` not concurrently safe, so it is necessary for the user a comment.